testing/jest.db-setup.ts

```diff
@@ -0,0 +1,49 @@
+import { beforeEach, afterAll, jest } from "@jest/globals";
+import { MongoMemoryServer } from "mongodb-memory-server";
+import * as Config from "../src/config";
+import mongoose from "mongoose";
+
+function mockConfig(dbUrl: string) {
+    jest.mock("../src/config.ts", () => {
+        const actual = jest.requireActual("../src/config.ts") as typeof Config;
+
+        const newConfig: typeof Config.default = {
+            ...actual.default,
+            DATABASE_HOST: dbUrl,
+            ENV: "TESTING",
+        };
+
+        return {
+            ...actual,
+            default: newConfig,
+            __esModule: true,
+        };
+    });
+}
+
+let mongod: MongoMemoryServer | undefined = undefined;
+
+beforeEach(async () => {
+    if (!mongod) {
+        mongod = await MongoMemoryServer.create();
+    }
+    const uri = `${mongod.getUri()}`;
+    if (mongoose.connections.length > 0) {
+        await mongoose.disconnect();
+    }
+    await mongoose.connect(
+        `${uri}?retryWrites=true&w=majority&appName=rp-dev-cluster`
+    );
+    mockConfig(`${uri}?retryWrites=true&w=majority&appName=rp-dev-cluster`);
+});
```

Comment:
are we creating the database before each test? seems like a lot of resource consumption

Reply:
imo:

beforeAll: spawn mongo inmem
afterAll: destroy mongo inmem

beforeEach: connect to server
afterEach: drop all the databases/remove connection


this way the overhead of spinning up server is only 1x per N tests, as opposed to Nx